### ISSUE TITLE
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting

### ISSUE THREAD

**Issue by omysurya** [maldua/zimbra-foss-builder#11]
[exec] Running [../zm-pkg-tool/pkg-build.pl --out-type=binary --pkg-name=zimbra-common-core-jar --pkg-version=10.1.3.1728626495 --pkg-release=1 --pkg-summary=Zimbra Core Jars --pkg-installs=/opt/zimbra/lib/jars/zimbracommon.jar --pkg-installs=/opt/zimbra/lib/jars/zimbrastore.jar --pkg-installs=/opt/zimbra/lib/jars/zimbra-native.jar --pkg-installs=/opt/zimbra/lib/jars/zimbraclient.jar --pkg-installs=/opt/zimbra/lib/jars/zimbrasoap.jar --pkg-replaces=zimbra-core --pkg-depends=zimbra-core-components]...
     [exec] =========================================================================================================
     [exec]  CFG_DIR                  : default           : pkg-spec
     [exec]  OUT_TYPE                 : cmdline           : binary
     [exec]  OUT_BASE_DIR             : default           : build
     [exec]  OUT_TEMP_DIR             : default           : build/tmp
     [exec]  OUT_STAGE_DIR            : default           : build/stage
     [exec]  OUT_DIST_DIR             : default           : build/dist
     [exec]  PKG_NAME                 : cmdline           : zimbra-common-core-jar
     [exec]  PKG_RELEASE              : cmdline           : 1
     [exec]  PKG_VERSION              : cmdline           : 10.1.3.1728626495
     [exec]  PKG_SUMMARY              : cmdline           : Zimbra Core Jars
     [exec]  PKG_INSTALLS             : cmdline           : [/opt/zimbra/lib/jars/zimbracommon.jar, /opt/zimbra/lib/jars/zimbrastore.jar, /opt/zimbra/lib/jars/zimbra-native.jar, /opt/zimbra/lib/jars/zimbraclient.jar, /opt/zimbra/lib/jars/zimbrasoap.jar]
     [exec]  PKG_DEPENDS              : cmdline           : [zimbra-core-components]
     [exec]  PKG_PRE_DEPENDS          : default           : []
     [exec]  PKG_PROVIDES             : default           : []
     [exec]  PKG_CONFLICTS            : default           : []
     [exec]  PKG_OBSOLETES            : default           : []
     [exec]  PKG_REPLACES             : cmdline           : [zimbra-core]
     [exec]  PKG_FORMAT               : default           : deb
     [exec]  PKG_OS_TAG               : default           : u20
     [exec] =========================================================================================================
     [exec] #: pwd=/opt/zimbra/zm-mailbox
     [exec] #: rm -rf build/tmp/zimbra-common-core-jar/
     [exec] Running [rm -rf build/tmp/zimbra-common-core-jar/]...
     [exec] #: pwd=/opt/zimbra/zm-mailbox
     [exec] #: mkdir -p build/tmp/zimbra-common-core-jar/
     [exec] Running [mkdir -p build/tmp/zimbra-common-core-jar/]...
     [exec] #: pwd=/opt/zimbra/zm-mailbox
     [exec] #: mkdir -p build/dist/u20/
     [exec] Running [mkdir -p build/dist/u20/]...
     [exec] #: pwd=/opt/zimbra/zm-mailbox
     [exec] #: cp -a /opt/zimbra/zm-pkg-tool/default-template/debian build/tmp/zimbra-common-core-jar/
     [exec] Running [cp -a /opt/zimbra/zm-pkg-tool/default-template/debian build/tmp/zimbra-common-core-jar/]...
     [exec] #: pwd=/opt/zimbra/zm-mailbox
     [exec] #: cp -a build/stage/zimbra-common-core-jar/opt build/tmp/zimbra-common-core-jar/opt
     [exec] Running [cp -a build/stage/zimbra-common-core-jar/opt build/tmp/zimbra-common-core-jar/opt]...
     [exec] #: pwd=/opt/zimbra/zm-mailbox/build/tmp/zimbra-common-core-jar
     [exec] #: dpkg-buildpackage -uc -us -b
     [exec] Running [dpkg-buildpackage -uc -us -b]...
     [exec] dpkg-buildpackage: info: source package zimbra-common-core-jar
     [exec] dpkg-buildpackage: info: source version 10.1.3.1728626495-1.u20
     [exec] dpkg-buildpackage: info: source distribution stable
     [exec] dpkg-buildpackage: info: source changed by Zimbra Builder (builder) <build@zimbra>
     [exec] dpkg-buildpackage: info: host architecture amd64
     [exec] dpkg-buildpackage: warning: debian/rules is not executable; fixing that
     [exec]  dpkg-source --before-build .
     [exec] dpkg-checkbuilddeps: error: Unmet build dependencies: debhelper (>= 9)
     [exec]
     [exec]
     [exec] =========================================================================================================
     [exec] FAILURE MSG : cmd='dpkg-buildpackage -uc -us -b'
     [exec] EXTRA INFO  : 'dpkg-buildpackage -uc -us -b' exited with value 3
     [exec]
     [exec] =========================================================================================================
     [exec] --Stack Trace--
     [exec] ../zm-pkg-tool/pkg-build.pl:660 called from main::System
     [exec] ../zm-pkg-tool/pkg-build.pl:699 called from main::Build
     [exec] ../zm-pkg-tool/pkg-build.pl:702 called from main::main
     [exec]
     [exec] =========================================================================================================
     [exec]
     [exec]
     [exec] =========================================================================================================
     [exec] FAILURE MSG : cmd='../zm-pkg-tool/pkg-build.pl --out-type=binary --pkg-name=zimbra-common-core-jar --pkg-version=10.1.3.1728626495 --pkg-release=1 --pkg-summary=Zimbra Core Jars --pkg-installs=/opt/zimbra/lib/jars/zimbracommon.jar --pkg-installs=/opt/zimbra/lib/jars/zimbrastore.jar --pkg-installs=/opt/zimbra/lib/jars/zimbra-native.jar --pkg-installs=/opt/zimbra/lib/jars/zimbraclient.jar --pkg-installs=/opt/zimbra/lib/jars/zimbrasoap.jar --pkg-replaces=zimbra-core --pkg-depends=zimbra-core-components'
     [exec] EXTRA INFO  : '../zm-pkg-tool/pkg-build.pl --out-type=binary --pkg-name=zimbra-common-core-jar --pkg-version=10.1.3.1728626495 --pkg-release=1 --pkg-summary=Zimbra Core Jars --pkg-installs=/opt/zimbra/lib/jars/zimbracommon.jar --pkg-installs=/opt/zimbra/lib/jars/zimbrastore.jar --pkg-installs=/opt/zimbra/lib/jars/zimbra-native.jar --pkg-installs=/opt/zimbra/lib/jars/zimbraclient.jar --pkg-installs=/opt/zimbra/lib/jars/zimbrasoap.jar --pkg-replaces=zimbra-core --pkg-depends=zimbra-core-components' exited with value 255
     [exec]
     [exec] =========================================================================================================
     [exec] --Stack Trace--
     [exec] ./pkg-builder.pl:538 called from main::System
     [exec] ./pkg-builder.pl:560 called from main::make_package
     [exec] ./pkg-builder.pl:575 called from main::depth_first_traverse_package
     [exec] ./pkg-builder.pl:664 called from main::main
     [exec]
     [exec] =========================================================================================================
     [exec] dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
     [exec] dpkg-buildpackage: warning: (Use -d flag to override.)
     [exec] END at ../zm-pkg-tool/pkg-build.pl line 146.
     [exec] END at ./pkg-builder.pl line 659.

BUILD FAILED
/opt/zimbra/zm-mailbox/build.xml:89: exec returned: 255

Total time: 5 minutes 49 seconds

=========================================================================================================
FAILURE MSG : cmd='ant -Ddebug=0 -Dis-production=1 -Dzimbra.buildinfo.platform=UBUNTU20_64 -Dzimbra.buildinfo.pkg_os_tag=u20 -Dzimbra.buildinfo.version=10.1.3_GA_4200000 -Dzimbra.buildinfo.type=FOSS -Dzimbra.buildinfo.release=20241120142317 -Dzimbra.buildinfo.date=20241120142317 -Dzimbra.buildinfo.host=mail.sentris.web.id -Dzimbra.buildinfo.buildnum=4200000 -DskipTests=true pkg-after-plough-through-tests'
EXTRA INFO  : 'ant -Ddebug=0 -Dis-production=1 -Dzimbra.buildinfo.platform=UBUNTU20_64 -Dzimbra.buildinfo.pkg_os_tag=u20 -Dzimbra.buildinfo.version=10.1.3_GA_4200000 -Dzimbra.buildinfo.type=FOSS -Dzimbra.buildinfo.release=20241120142317 -Dzimbra.buildinfo.date=20241120142317 -Dzimbra.buildinfo.host=mail.sentris.web.id -Dzimbra.buildinfo.buildnum=4200000 -DskipTests=true pkg-after-plough-through-tests' exited with value 1

=========================================================================================================
--Stack Trace-- (37098)
./build.pl:555 called from main::SysExec
./build.pl:1018 called from main::__ANON__
./build.pl:577 called from main::RunInDir
./build.pl:1080 called from main::Build
./build.pl:1086 called from main::main

=========================================================================================================
END at ./build.pl line 1062.

=========================================================================================================
FAILURE MSG : child 37098 died
EXTRA INFO  : ret=255

=========================================================================================================
--Stack Trace-- (35929)
./build.pl:577 called from main::RunInDir
./build.pl:1080 called from main::Build
./build.pl:1086 called from main::main

=========================================================================================================
END at ./build.pl line 1062.

**Comment by adriangibanelbtactic** [comment]
Please enclose big texts from output thanks to three backticks. '```'. Check [github markdown](https://docs.github.com/es/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).

Also when reporting a new bug explain in plain English:

- What you were trying to do
- What you were expecting
- What happened instead

.

Thank you very much!

**Comment by adriangibanelbtactic** [comment]
If you want help on building Zimbra thanks to zimbra-foss-builder you should specify in which OS you are building it and what commands have you run to do so.

Also check the [current documentation](https://github.com/maldua/zimbra-foss-builder?tab=readme-ov-file#smart-build-example-recommended).

**Comment by adriangibanelbtactic** [comment]
> [exec] dpkg-checkbuilddeps: error: Unmet build dependencies: debhelper (>= 9)

If you are going for the non-documented **manual build without Docker help** you should be checking what the [specific Dockerfile for your OS](https://github.com/maldua/zimbra-foss-builder/blob/e5ef74357504b44751cc7638a99b9891058cbd35/Dockerfile-manual-ubuntu-20.04) is doing and replicate that manually.

As you can see as part of setting up the environment the **debhelper** package (with many more other packages) is installed.

Your build is complaining about debhelper not being there in the first place.

In any case I'm waiting for your system exact description (Distro version) and the exact commands that you ran, just in case I have misunderstood what you were actually doing.

**Comment by adriangibanelbtactic** [comment]
No feedback in a long time.

Closing the issue.
